src/transitions/page_transitions.rs

```diff
@@ -43,381 +45,65 @@ impl<R: Routable + PartialEq> AnimatedRouterContext<R> {
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-            TransitionVariant::SpiralIn => TransitionConfig {
-                initial_from: Transform::new(0.0, 0.0, 1.0, 1.0),
-                final_from: Transform::new(0.0, 0.0, 0.0, -720.0),
-                initial_to: Transform::new(0.0, 0.0, 2.0, 0.0),
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-            TransitionVariant::SpiralOut => TransitionConfig {
-                initial_from: Transform::new(0.0, 0.0, 1.0, 1.0),
-                final_from: Transform::new(0.0, 0.0, 2.0, 720.0),
-                initial_to: Transform::new(0.0, 0.0, 0.0, 0.0),
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-            TransitionVariant::ElasticIn => TransitionConfig {
-                initial_from: Transform::new(0.0, 0.0, 1.0, 1.0),
-                final_from: Transform::new(0.0, 0.0, 0.3, 0.0),
-                initial_to: Transform::new(0.0, 30.0, 1.5, 0.0),
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-            TransitionVariant::ElasticOut => TransitionConfig {
-                initial_from: Transform::new(0.0, 0.0, 1.0, 1.0),
-                final_from: Transform::new(0.0, 30.0, 1.5, 0.0),
-                initial_to: Transform::new(0.0, 0.0, 0.3, 0.0),
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-            TransitionVariant::SwingIn => TransitionConfig {
-                initial_from: Transform::new(0.0, 0.0, 1.0, 1.0),
-                final_from: Transform::new(0.0, 0.0, 1.0, -20.0),
-                initial_to: Transform::new(0.0, 0.0, 1.0, 20.0),
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-            TransitionVariant::SwingOut => TransitionConfig {
-                initial_from: Transform::new(0.0, 0.0, 1.0, 1.0),
-                final_from: Transform::new(0.0, 0.0, 1.0, 20.0),
-                initial_to: Transform::new(0.0, 0.0, 1.0, -20.0),
-                final_to: Transform::new(0.0, 0.0, 1.0, 1.0),
-            },
-        }
-    }
-}
-
 #[component]
-fn FromRouteToCurrent<R: AnimatableRoute>(
-    route_type: PhantomData<R>,
-    from: Element,
-    transition: TransitionVariant,
-) -> Element {
+fn FromRouteToCurrent<R: AnimatableRoute>(route_type: PhantomData<R>, from: R, to: R) -> Element {
     let mut animated_router = use_animated_router::<R>();
-    let config = transition.get_config();
+    let config = to.get_transition().get_config();
     let mut from_transform = use_motion(config.initial_from);
     let mut to_transform = use_motion(config.initial_to);
     let mut from_opacity = use_motion(1.0f32);
     let mut to_opacity = use_motion(0.0f32);
 
+    let outlet: OutletContext<R> = use_outlet_context();
+    let current_level = outlet.current_level;
+
+    provide_context({
+        OutletContext::<R> {
+            current_level: current_level + 1,
```

Comment:
This should probably be in a `use_hook` or changed to `use_context_provider` so you don't increase the outlet level whenever the component reruns